tests/isolated/test_sub_module_in_import_patched/test.py

```diff
@@ -0,0 +1,10 @@
+import eventlet
+import tests.isolated.test_sub_module_in_import_patched.sample_main_module as test_module
+
+
+if __name__ == '__main__':
+    assert test_module.function_use_socket() is not eventlet.green.socket
```

Comment:
One last nitpick – can you move this *after` the `import_patched` call? We're mostly interested if `import_patched` doesn't break things.

Reply:
Sure:
* this assert doesn't check anything useful right now. Why would the returned object be anything other than the standard `socket` module?
* `import_patched` can potentially mutate modules that it should not mutate, therefore moving the assert *after* the call makes sure the non-monkey-patched module is ok

```python
    patched_module = eventlet.import_patched(
        'tests.isolated.test_sub_module_in_import_patched.sample_main_module')
    assert patched_module.function_use_socket() is eventlet.green.socket
    assert test_module.function_use_socket() is not eventlet.green.socket
```

Reply:
That makes sense. I intended to test the net effect of import_patched is changing the socket symbol in __globals__. But this is tested with the assert moving to the end as well. Added the change. PTAL.